decision/decision_test.go

```diff
@@ -237,3 +237,36 @@ func TestDecisionBucketInNoCache(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Len(t, decision.Campaigns, 1)
 }
+
+func TestVisitorShouldNotBeAssignedWhenVariationDeleted(t *testing.T) {
+	vi := VisitorInfo{}
+	vi.ID = "v1"
+	vi.DecisionGroup = "dg"
+	vi.Context = map[string]*structpb.Value{
+		"isVIP": structpb.NewBoolValue(true),
+	}
+
+	ei := EnvironmentInfo{}
+	ei.ID = "e123"
+	ei.Campaigns = campaigns
+	for _, vg := range ei.Campaigns["a"].VariationsGroups {
+		vg.Campaign = ei.Campaigns["a"]
+	}
+
+	options := DecisionOptions{
+		TriggerHit: true,
+	}
+	// no options
+
+	handlers := DecisionHandlers{}
+	handlers.GetCache = mockGetCache
+	handlers.SaveCache = mockSaveCache
+	handlers.ActivateCampaigns = mockActivateCampaigns
+
+	// delete variation and check that visitor is not returned
+	campaignVars := ei.Campaigns["a"].VariationsGroups["vga"].Variations
+	ei.Campaigns["a"].VariationsGroups["vga"].Variations = append(campaignVars[:0], campaignVars[1:]...)
```

Comment:
campaignVars[:0] is empty.
`ei.Campaigns["a"].VariationsGroups["vga"].Variations` = campaignVars[1:]` would be better

Reply:
done thanks :)